### ISSUE TITLE
ATC firmware and errors with data length

### ISSUE THREAD

**Issue by GrzegorzZajac000** [custom-components/ble_monitor#315]
I have 4 Xiaomi LYWSD03MMC with custom ATC firmware and I can't add it to my HomeAssistant. I updated ble_monitor to 1.4 but problem still exists.

Logs:
```2021-03-30 10:26:54 WARNING (MainThread) [homeassistant.loader] You are using a custom integration hacs which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-03-30 10:26:54 WARNING (MainThread) [homeassistant.loader] You are using a custom integration ble_monitor which has not been tested by Home Assistant. This component might cause stability problems, be sure to disable it if you experience issues with Home Assistant
2021-03-30 10:26:57 DEBUG (MainThread) [custom_components.ble_monitor] Initializing BLE Monitor integration (YAML): {'hci_interface': [0], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': [{'mac': 'A4:C1:38:BC:7C:4E', 'name': 'Czujnik Salon', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:63:37:8B', 'name': 'Czujnik Sypialnia', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:D2:55:BD', 'name': 'Czujnik Biuro', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:4E:6A:1B', 'name': 'Czujnik Łazienka', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}], 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:26:59 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_import: {'hci_interface': [0], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': [{'mac': 'A4:C1:38:BC:7C:4E', 'name': 'Czujnik Salon', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:63:37:8B', 'name': 'Czujnik Sypialnia', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:D2:55:BD', 'name': 'Czujnik Biuro', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:4E:6A:1B', 'name': 'Czujnik Łazienka', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}], 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:26:59 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] async_step_user: {'hci_interface': [0], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': '--Devices--', 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:26:59 DEBUG (MainThread) [custom_components.ble_monitor.config_flow] _create_entry: {'hci_interface': [0], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': '--Devices--', 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] Initializing BLE Monitor entry (config entry): <homeassistant.config_entries.ConfigEntry object at 0x7fdeac23a360>
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] async_setup_entry: domain {'hci_interface': [0], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': [{'mac': 'A4:C1:38:BC:7C:4E', 'name': 'Czujnik Salon', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:63:37:8B', 'name': 'Czujnik Sypialnia', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:D2:55:BD', 'name': 'Czujnik Biuro', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:4E:6A:1B', 'name': 'Czujnik Łazienka', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}], 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] async_setup_entry: {'hci_interface': ['0'], 'discovery': True, 'active_scan': True, 'report_unknown': True, 'batt_entities': True, 'decimals': 1, 'period': 60, 'log_spikes': False, 'use_median': False, 'restore_state': False, 'devices': [{'mac': 'A4:C1:38:BC:7C:4E', 'name': 'Czujnik Salon', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:63:37:8B', 'name': 'Czujnik Sypialnia', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:D2:55:BD', 'name': 'Czujnik Biuro', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}, {'mac': 'A4:C1:38:4E:6A:1B', 'name': 'Czujnik Łazienka', 'temperature_unit': '°C', 'decimals': 1, 'use_median': False, 'restore_state': 'default', 'reset_timer': 35}], 'rounding': 1, 'is_flow': False, 'ids_from_name': True}
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] HCI interface is [0]
2021-03-30 10:27:00 INFO (MainThread) [custom_components.ble_monitor] Attention! Option report_unknown is enabled, be ready for a huge output...
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] Spawning HCIdump thread
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] HCIdump thread: Init
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] Attention! Option report_unknown is enabled, be ready for a huge output...
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] 0 encryptors mac:key pairs loaded.
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] whitelist: []
2021-03-30 10:27:00 DEBUG (MainThread) [custom_components.ble_monitor] 0 whitelist item(s) loaded.
2021-03-30 10:27:00 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:27:00 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:27:00 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Starting measuring sensor entry startup
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialization
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] BLE sensors updater initialized
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Measuring sensor entry setup finished
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Starting binary sensor entry startup
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialization
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] BLE binary sensors updater initialized
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary sensor entry setup finished
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Entities updater loop started!
2021-03-30 10:27:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary entities updater loop started!
2021-03-30 10:27:01 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:27:02 DEBUG (MainThread) [custom_components.ble_monitor.sensor] Sensor device with mac address AC:9A:22:F1:A9:D1 has the following settings. Name: AC9A22F1A9D1. Temperature unit: °C. Decimals: 1. Use Median: False. Restore state: False. Reset Timer: 35.
2021-03-30 10:27:02 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] Binary sensor device with mac address AC:9A:22:F1:A9:D1 has the following settings. Name: AC9A22F1A9D1. Restore state: False. Reset Timer: 35
2021-03-30 10:27:03 DEBUG (MainThread) [custom_components.ble_monitor.sensor] async_added_to_hass called for ble temperature AC9A22F1A9D1
2021-03-30 10:27:03 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] async_added_to_hass called for ble switch AC9A22F1A9D1
2021-03-30 10:27:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:27:08 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:28:01 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:28:01 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:28:01 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:28:01 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:28:01 DEBUG (Thread-5) [custom_components.ble_monitor] 43 HCI events processed for previous period.
2021-03-30 10:28:01 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:28:01 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:28:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:28:03 ERROR (MainThread) [homeassistant.components.google_assistant.http] Request for https://homegraph.googleapis.com/v1/devices:reportStateAndNotification failed: 404
2021-03-30 10:28:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:28:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:28:06 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:29:02 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:29:02 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] 45 HCI events processed for previous period.
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:29:02 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:29:04 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:30:02 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:30:02 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] 43 HCI events processed for previous period.
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:30:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:30:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:30:06 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:30:10 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:31:02 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:31:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:31:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:31:02 DEBUG (Thread-5) [custom_components.ble_monitor] 46 HCI events processed for previous period.
2021-03-30 10:31:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:31:02 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:31:02 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:31:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:31:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:31:03 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:31:22 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:32:03 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:32:03 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] 42 HCI events processed for previous period.
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:32:03 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:32:09 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:32:15 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:32:22 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:33:04 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 1 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 10:33:04 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 1 MiBeacon BLE ADV messages processed for 1 measuring device(s).
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] 45 HCI events processed for previous period.
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 10:33:04 DEBUG (Thread-5) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 10:33:05 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:33:16 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 10:33:22 DEBUG (Thread-5) [custom_components.ble_monitor] Invalid data: Invalid data length```

Added Xiaomi Beacon is my MiKettle ;)

**Comment by Ernst79** [comment]
Which ATC firmware version do you have? And what advertisement type do you use in the ATC settings?

and try to to set active scan to false (=default) and report_unknown can also be set to false.

**Comment by GrzegorzZajac000** [comment]
ATC firmware Release 34. I installed it yesterday. I didn't change Advertisement type so I probably have default for ATC.
I changed config in YAML, but nothing changed in HA

**Comment by Ernst79** [comment]
Release 34? Ah, you use this atc1441 version (https://github.com/atc1441/ATC_MiThermometer)

I normally use this improved pvvx ATC version, https://github.com/pvvx/ATC_MiThermometer, which is currently at version 2.9. 

I'll check tonight if there has been a change. The 1.4.0 version was changed due to the v2.9 update of the last firmware. It should be backward compatible with the old pvvx version <2.8 and also with the atc1441 version v34 you use, but I'll check tonight if I accidently broke something

**Comment by GrzegorzZajac000** [comment]
Yes, I am using atc1441 version. Thank you :)

**Comment by Ernst79** [comment]
I just tried it myself with V34, don't see an issue on my end. 

What kind of Home Assistant installation do you have? If you have a supervised installation, venv installation or docker installation, I would like to ask you to make a HCI dump with this command

```
sudo hcidump --raw hci > dump.txt
```

Collect for about 2 minutes of data, and upload this file to github by attaching it to a message. 

This doesn't work easy on a Home Assistant installation that includes Home Assistant OS (the full version you download to an SD card). If you have this version, let me know. I have an other way to get the data, but we have to modify the code than temporary.

**Comment by GrzegorzZajac000** [comment]
I have HomeAssistant installed by Whiskerz007 script on Proxmox. I use Dell Wyse 5070 and built-in Intel 9560 PCI card.
And my false, but I tried connect to BT sensor for couple days and I have become impatient and I decided to reflash all my sensors to pvvx and everything works fine.

If you want I can re-flash one of my sensors, but dump.txt for now.
[dump.txt](https://github.com/custom-components/ble_monitor/files/6228703/dump.txt)

**Comment by Ernst79** [comment]
At least, you now know a way to get it working. :-)

To help the community, it would be really appreciated if you can flash it (temporary) with atc1441 firmware and generate a HCI dump again. This will help me fix it for other users.

**Comment by GrzegorzZajac000** [comment]
Here you are
[dump.txt](https://github.com/custom-components/ble_monitor/files/6228804/dump.txt)

I checked logs. Invalid data length appeared

**Comment by Ernst79** [comment]
Are you sure the sensor isn't working? I don't see an issue in the format of the messages.

**Comment by GrzegorzZajac000** [comment]
I think so, because of the logs:
```2021-03-30 13:55:06 DEBUG (MainThread) [custom_components.ble_monitor.binary_sensor] 0 MiBeacon BLE ADV messages processed for 1 binary sensor device(s) total. Priority queue = 0
2021-03-30 13:55:18 DEBUG (MainThread) [custom_components.ble_monitor.sensor] 3 MiBeacon BLE ADV messages processed for 5 measuring device(s).
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] HCIdump thread: main event_loop stopped, finishing
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] HCIdump thread: Scanning will be restarted
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] 13 HCI events processed for previous period.
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] HCIdump thread: Run
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] HCIdump thread: connected to hci0
2021-03-30 13:55:18 DEBUG (Thread-13) [custom_components.ble_monitor] HCIdump thread: start main event_loop
2021-03-30 13:55:23 DEBUG (Thread-13) [custom_components.ble_monitor] Invalid data: Invalid data length
2021-03-30 13:55:23 DEBUG (Thread-13) [custom_components.ble_monitor] Invalid data: Invalid data length```

**Comment by GrzegorzZajac000** [comment]
And I didn't get any update on this sensor since re-flash

**Comment by Ernst79** [comment]
ok, I'll look further. Please keep the firmware for a few hours. 

I'm now looking at a message of the sensor with mac a4c13863378b. Is that a sensor that isn't working?

**Comment by GrzegorzZajac000** [comment]
Yes, I reflashed only one sensor with MAC: A4:C1:38:BC:7C:4E

**Comment by Ernst79** [comment]
I think the HCI dump is too short, there is no message from this sensor. Can you set the interval to 10 seconds in the firmware settings and/or make a longer hcidump? I thought you can set the interval to 10 seconds on the flashing website.

**Comment by Ernst79** [comment]
Just a side note, the invalid data length isn't necessarily an issue. Sensors often send data that are not useful, it is just a debug message that tells me that this message is skipped. As long as there are enough other messages that do work, this should be a problem. 

But a sensor not updating is an issue.

**Comment by GrzegorzZajac000** [comment]
[dump.txt](https://github.com/custom-components/ble_monitor/files/6228993/dump.txt)

**Comment by GrzegorzZajac000** [comment]
And no values update from this reflashed sensor

**Comment by Ernst79** [comment]
Thanks, I found a BLE message of the sensor.

**Comment by GrzegorzZajac000** [comment]
![Screenshot 2021-03-30 at 16 14 32](https://user-images.githubusercontent.com/12233796/113003442-0a007200-9173-11eb-9a3f-97bcb39baf14.png)

It works ;)

**Comment by Ernst79** [comment]
Great, thanks. Update will be released tonight. You can return to ATC pvvx if you want. Thanks for your help.

**Comment by Ernst79** [comment]
Fixed in 1.4.1
